src/main/java/org/junit/Assert.java

```diff
@@ -812,6 +812,37 @@ public static void assertNotSame(Object unexpected, Object actual) {
         assertNotSame(null, unexpected, actual);
     }
 
+    /**
+     * Asserts that two {@link CharSequence} instances have the same contentsa.
+     * If they do not, an {@link AssertionError} is thrown.
+     *
+     * @param expected the expected contents
+     * @param actual the object to compare to
+     */
+    public static void assertContentEquals(CharSequence expected, CharSequence actual)  {
+        assertContentEquals(null, expected, actual);
+    }
+
+    /**
+     *Asserts that two {@link CharSequence} instances have the same contentsa.
+     * If they do not, an {@link AssertionError} is thrown with the given message.
+     *
+     * @param message the identifying message for the {@link AssertionError}
+     * @param expected the expected contents
+     * @param actual the object to compare to
+     */
+    public static void assertContentEquals(String message, CharSequence expected, CharSequence actual) {
+        if (expected == null && actual == null) {
+            return;
+        }
+        if (expected == null || actual == null) {
+            failNotEquals(message, expected, actual);
+        }
+
+        assertEquals(expected.length(), actual.length());
```

Comment:
Actually, I suggested comparing the lengths first only to avoid calling `toString()`.
But since `assertEquals(..., actual.toString())` is used for comparing the content there is no point to do a separate length check (`String.equals()` will do it anyway).

Besides, not checking the lengths (via `assertEquals`) would result in a nicer message. Compare:
> Expected: <5> but was: <2>

vs

> Expected: "Hello" but was: "Hi"